src/lib/metrics.js

```diff
@@ -0,0 +1,340 @@
+// Use of package.json for configuration
+import packageMeta from '../../package.json';
+
+import { makeLog } from './utils';
+import { selectors } from './store';
+import { colorsWithAlpha } from './constants';
+
+const log = makeLog('metrics');
+
+const GA_URL = 'https://ssl.google-analytics.com/collect';
+/* eslint-disable prefer-destructuring */
+const GA_TRACKING_ID = packageMeta.config.GA_TRACKING_ID;
+const GA_APP_ID = packageMeta.extensionManifest.applications.gecko.id;
+const GA_APP_VERSION = packageMeta.version;
+/* eslint-enable prefer-destructuring */
+
+let clientUUID = null;
+
+// Time after which to consider add-on install to have failed, because we have
+// no good way to tell if the install has been cancelled - only if the add-on
+// becomes available
+const INSTALL_FAILURE_DELAY = 10 * 1000;
+let installTimer = null;
+
+// See also: https://github.com/mozilla/ThemesRFun/blob/master/docs/metrics.md
+
+// Custom metrics
+let cm1; // integer count of select-full events from start of visit
+let cm2; // integer count of select-bg events from start of visit
+let cm3; // integer count of select-color events from start of visit
```

Comment:
These aren't going to be the most useful of custom metrics, since looking for a sum is meaningless (a user who got to `cm3=3` will have a `cm3` sum of `6`), and thus means are also not...meaningful.

Reply:
I think that's more a concern for #11 and metrics.md - this PR is just implementing what we have so far, which we can iterate on if we revise the schema